### PR TITLE
refactor: migrate number validator to typescript

### PR DESCRIPTION
**Pull request by tshuli**
## Problem
Part of #7

## Solution

**Improvements**

- Migrates number validation to Typescript

## Tests
- [ ] Create two number fields. Hide one of them behind logic.
- [ ] Check that validation on both visible and hidden number fields work when submitting programatically
- Response is rejected if it contains non numeric characters
- Response is rejected if not within the specified length

**Unit test coverage**

Number field validation
    ✓ should allow number with valid minimum (inclusive)
    ✓ should allow number with valid maximum (inclusive)
    ✓ should allow number with valid exact
    ✓ should disallow number with invalid maximum
    ✓ should disallow number with invalid exact
    ✓ should allow leading zeroes in answer
    ✓ should allow number with valid maximum
    ✓ should disallow responses submitted for hidden fields
    ✓ should disallow negative answers
    ✓ should allow number with exact undefined
    ✓ should allow number with no custom validation
    ✓ should allow number with maximum left undefined
    ✓ should allow number with valid minimum
    ✓ should allow number with minimum left undefined
    ✓ should allow number with optional answer
    ✓ should allow answer to be zero

### CODE REVIEW

File: src/app/utils/field-validation/validators/numberValidator.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+const getNumberLengthValidator: NumberValidatorConstructor = (numberField) => {
+  switch (numberField.ValidationOptions.selectedValidation) {
+    case NumberSelectedValidation.Min:
+      return minLengthValidator(numberField)
+    case NumberSelectedValidation.Max:
+      return maxLengthValidator(numberField)
+    case NumberSelectedValidation.Exact:
+      return exactLengthValidator(numberField)
+    default:
```
Comment (karrui) [inline src/app/utils/field-validation/validators/numberValidator.ts:63]: ~~Should there also be a `NumberSelectedValidation.Range` case? Seems like it was not in the old code but should also be handled here.~~

EDIT: Just checked the code; `.Range` is not used anywhere, maybe we can delete that enum actually, if it does not exist in the DB. 

EDIT 2: Seems like there already is an issue tracked at https://github.com/opengovsg/FormSG/issues/408

File: tests/unit/backend/utils/field-validation/number-validation.spec.js [hunk 2]
```diff
@@ -235,6 +235,8 @@ describe('Number field validation', () => {
       answer: '55',
     }
     const validateResult = validateField('formId', formField, response)
+    console.log('############')
```
Comment (karrui) [inline tests/unit/backend/utils/field-validation/number-validation.spec.js:238]: Remove console logs?

Comment (tshuli) [reply tests/unit/backend/utils/field-validation/number-validation.spec.js]: done, thanks for pointing out

File: src/app/utils/field-validation/validators/numberValidator.ts (hunk 1 above)
Comment (tshuli) [reply src/app/utils/field-validation/validators/numberValidator.ts:63]: Thanks! Discussed at syncup and we'll port over the exact implementation first so we don't change too many things at once, and deal with #408 later

**Review by karrui — APPROVED** [review]
lgtm

File: src/app/utils/field-validation/validators/numberValidator.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+import { NumberSelectedValidation } from '../../../../types/field'
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type NumberValidator = ResponseValidator<ProcessedSingleAnswerResponse>
+type NumberValidatorConstructor = (numberField: INumberField) => NumberValidator
+
+/**
+ * Validator, to check that answer is an empty string or a nonnegative integer
```
Comment (mantariksh) [inline src/app/utils/field-validation/validators/numberValidator.ts:16]: nit: I find it useful to start function docs with a verb, and also to try not to repeat information which is already apparent from the function name
```suggestion
 * Checks that answer is an empty string or a nonnegative integer
```

**Review by mantariksh — APPROVED** [review]
could you check that the unit tests cover the edge case that the value is 0?
